src/H5Gdeprec.c

```diff
@@ -1075,23 +1075,17 @@ H5G__get_objinfo(const H5G_loc_t *loc, const char *name, bool follow_link, H5G_s
         HGOTO_ERROR(H5E_SYM, H5E_EXISTS, FAIL, "name doesn't exist");
 
     /* If we're pointing at a soft or UD link, get the real link length and type */
-    if (statbuf && follow_link == 0) {
```

Comment:
That will be handled by the H5G__iterate() routine internally and won't fail, since there's an actual object that is found during path iteration.   This case is only for the literal "." string.

Reply:
Hah!  I was wrong, good call @fortnern.  :-)

I've updated the code and added a test for this case.